packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ContainerRuntimeMessage } from "..";
+import { IBatch } from "./definitions";
+
+interface IGroupedMessage {
+	contents?: any;
+	metadata?: Record<string, unknown>;
```

Comment:
Given that we assert there could be no metadata on ops, it feels like we should remove it from here.

Reply:
We need the batch metadata for the `opDecompressor`